### ISSUE TITLE
✨ Add support for `notActions` and `notDataActions` to RBAC rule

### ISSUE THREAD

**Issue by mattwelke** [validator-labs/validator-plugin-azure#209]
### Summary

As part of an effort throughout the whole validator project, we are trying to re-use structs defined in 3rd party libraries related to plugins as much as possible.

This prefers to phase 2 of 2 of an effort to do this for Azure. After phase 1, we will be re-using the permissions struct (`Permissions` in `github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v2`) for our rule spec, but `notActions` and `notDataActions`, hereafter referred to as "not actons", would be ignored because we don't support them yet.

We should add support for not actions.

Right now, the RBAC rule includes a list of permissions the principal should have at a particular scope. The list of permissions can include wildcards, but it doesn't allow any not actions. Therefore, if a user wanted to subtract a permission from a permission includes a wildcard (e.g. remove `Microsoft.Authorization/*/Delete` from `*`), they would have to change the permission with a wildcard to the expanded out version of it that does not use a wildcard (e.g. change `*` to all of the `Microsoft.Authorization` permissions except for `Microsoft.Authorization/*/Delete`).

Being able to use not actions would make it easier for them to use the plugin because they wouldn't have to expand out the permissions they define this way. This is the same approach that Azure uses with not actions in their APIs to make it easier for their users to use the APIs too.

Doing this will probably require us to expand out `actions` and `notActions` with wildcards on our side and then remove the not actions from them, similar to what Azure is doing on their side. This may not be possible because we may not have access to the expanded out list of permissions programatically. This needs research.

**Comment by mattwelke** [comment]
Closing in favor of https://github.com/validator-labs/validator-plugin-azure/issues/215, which will support wildcards, notActions, and notDataActions.
